pkg/flare/archive_test.go

```diff
@@ -506,3 +509,75 @@ func TestZipVersionHistory(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, "mockfilecontent", string(actualContent))
 }
+
+func TestZipProcessAgentFullConfig(t *testing.T) {
+	type ProcessConfig struct {
+		Enabled string `yaml:"enabled"`
+	}
+
+	globalCfg := struct {
+		Apikey     string        `yaml:"api_key"`
+		DDurl      string        `yaml:"dd_url"`
+		ProcessCfg ProcessConfig `yaml:"process_config"`
+	}{
+		"*****1234",
+		"https://my-url.com",
+		ProcessConfig{
+			"true",
+		},
+	}
+
+	exp := `api_key: '*****1234'
+dd_url: https://my-url.com
+process_config:
+  enabled: "true"`
+
+	cfg := config.Mock()
+	// Use different port in case the host is running a real agent
+	cfg.Set("process_config.cmd_port", 8182)
+
+	t.Run("without process-agent running", func(t *testing.T) {
+		dir, err := ioutil.TempDir("", "TestZipProcessAgentFullConfig")
+		require.NoError(t, err)
+		defer os.RemoveAll(dir)
+
+		zipProcessAgentFullConfig(dir, "")
+		content, err := ioutil.ReadFile(filepath.Join(dir, "process_agent_runtime_config_dump.yaml"))
+		require.NoError(t, err)
+		assert.Equal(t, "error: process-agent is not running or is unreachable", string(content))
+	})
+
+	t.Run("with process-agent running", func(t *testing.T) {
+		// Create a server to mock process-agent /config/all endpoint
+		cfgMux := http.NewServeMux()
+		cfgMux.HandleFunc("/config/all", func(w http.ResponseWriter, _ *http.Request) {
+			b, err := yaml.Marshal(globalCfg)
+			require.NoError(t, err)
+
+			_, err = w.Write(b)
+			require.NoError(t, err)
+		})
+
+		cfgEndpoint := fmt.Sprintf("localhost:%d", cfg.GetInt("process_config.cmd_port"))
+		cfgSrv := http.Server{Addr: cfgEndpoint, Handler: cfgMux}
```

Comment:
btw, `httptest` can make this all a bit easier (no need to change it here, just FYI)

In particular, it allows the test to run with an arbitrary port, rather than a hard-coded port.

Reply:
Oh! Thanks for pointing that out, that's really useful knowledge for our team 😄

Reply:
Thanks for pointing it out, @djmitche. I refactored a bit the code and updated the test to use `httptest`. It's much cleaner now!